lib/event_listeners.js

```diff
@@ -469,14 +469,52 @@ AWS.EventListeners = {
       var req = resp.request;
       var logger = req.service.config.logger;
       if (!logger) return;
+      function filterSensitiveLog(inputShape, shape) {
+        switch (inputShape.type) {
```

Comment:
I think you need to return early here if `shape` is undefined. There are recursive shapes defined in some services, and this function will need to check for some stopping condition.

Reply:
Oh right. Thank you for pointing out.